src/main/java/Duke.java

```diff
@@ -1,10 +1,109 @@
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+
+import javafx.application.Platform;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    private Storage storage;
+    private TaskList tasks;
+    private Gui ui;
+
+    /**
+     * Constructor for the Duke Class.
+     * <p>
+     * //@param filepath Path to file where the data for saved tasks are stored.
+     */
+    public Duke() {
+        String filepath = "data/duke.txt";
+        ui = new Gui();
+        //ui.printStartUp();
+        storage = new Storage(filepath);
+        try {
+            tasks = new TaskList(storage.loadData());
+        } catch (FileNotFoundException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+
+
+    /*public static void main(String[] args) {
+        new Duke().run();
+    }*/
+
+    /**
+     * Runs the code for the Duke program.
+     */
+    public String getResponse(String in) {
+        String response = "";
+        Parser parser = new Parser(in);
+        if (parser.getCommand().equals("bye")) {
+            response = (ui.terminateGui());
+            Platform.exit();
+        } else if (parser.getCommand().equals("list")) {
+            response = (ui.printOutTasks(tasks));
+        } else if (parser.getCommand().equals("delete")) {
+            try {
+                response = ui.printOutDeleted(tasks, parser.getIndex() - 1);
+                tasks.removeTask(parser.getIndex() - 1);
+                //return toReturn;
+            } catch (DukeException e) {
+                response = e.toString();
+            }
+            try {
+                storage.saveData(tasks);
+            } catch (IOException e) {
+                response += ("\nAn error occurred while saving, please try again!\n");
+            }
+        } else if (parser.getCommand().equals("done")) {
+            try {
+                tasks.getTask(parser.getIndex() - 1).doTask();
+                response = (ui.printOutDoneTask(tasks, parser.getIndex() - 1));
+            } catch (DukeException e) {
+                response = e.toString();
+            }
+            try {
+                storage.saveData(tasks);
+            } catch (IOException e) {
+                response += ("\nAn error occurred while saving, please try again!\n");
+            }
```

Comment:
Repeated catch blocks. Can try putting the try catch block outside and throw exception with different message instead.